v3/lints/cabf_br/lint_invalid_ca_certificate_policies.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * ZLint Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * Contributed by asantoni64@gmail.com
+ */
+
+package cabf_br
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_invalid_ca_certificate_policies",
+			Description:   "Checks compliance with CABF section 7.1.2.10.5",
+			Citation:      "CABF BRs §7.1.2.10.5",
+			Source:        lint.CABFBaselineRequirements,
+			EffectiveDate: util.CABFBRs_2_0_0_Date,
+		},
+		Lint: NewInvalidCACertificatePolicies,
+	})
+}
+
+type invalidCACertificatePolicies struct{}
+
+func NewInvalidCACertificatePolicies() lint.LintInterface {
+	return &invalidCACertificatePolicies{}
+}
+
+func (l *invalidCACertificatePolicies) CheckApplies(c *x509.Certificate) bool {
+	return util.IsCACert(c) && !util.IsRootCA(c)
+}
+
+func (l *invalidCACertificatePolicies) Execute(c *x509.Certificate) *lint.LintResult {
+	if !util.IsExtInCert(c, util.CertPolicyOID) {
+		return &lint.LintResult{
+			Status:  lint.Error,
+			Details: "In a non-Root CA certificate, the CertificatePolicies extension is mandatory",
+		}
+	}
+
+	if len(c.PolicyIdentifiers) == 1 {
+		if !c.PolicyIdentifiers[0].Equal(util.AnyPolicyOID) {
+			return &lint.LintResult{
+				Status:  lint.Error,
+				Details: "If a CA cert contains only one policy OID, then it MUST be AnyPolicy",
```

Comment:
I'm not seeing this specific text in the BR. Do you have a pointer?

Reply:
Sure. 

First of all, 7.1.2.10.5 reads like follows in the case of un-restricted CAs (No Policy Restrictions):
> When the Issuing CA wishes to express that there are no policy restrictions, the Subordinate CA MUST be an Affiliate of the Issuing CA. The Certificate Policies extension MUST contain only a **single** PolicyInformation value, which MUST contain the **anyPolicy** Policy Identifier.

In the case of Policy-Restricted CAs, instead, the same section reads:

> The CA MUST include **at least one Reserved** Certificate Policy Identifier (see Section 7.1.6.1) associated with the given Subscriber Certificate type (see Section 7.1.2.7.1) directly or transitively issued by this Certificate ... [and] ...
> The **anyPolicy** Policy Identifier **MUST NOT** be present.

This amounts to saying, if I am not mistaken, that if CertificatePolicies contains just one OID then it must be anyPolicy; otherwise, _at least one_ of the included Policy OIDs must be a CABF reserved OID and _none_ of them can be anyPolicy (because this latter must be alone per the first rule in §7.1.2.10.5).

Reply:
I realized that my conclusion is wrong, because in fact it is allowed that the CP extension contains only one Policy OID as long as, in that case, it is a CABF reserved one. I was a bit too hasty, but in these things one must be more thoughtful :) 

I rewrote and retested my code, also adding more comments and one more test case.